contrib/IECoreUSD/src/IECoreUSD/PrimitiveAlgo.cpp

```diff
@@ -103,6 +112,47 @@ void IECoreUSD::PrimitiveAlgo::writePrimitiveVariable( const std::string &name,
 	}
 }
 
+namespace
+{
+
+struct VtValueFromExpandedData
+{
+
+	template<typename T>
+	VtValue operator()( const IECore::TypedData<vector<T>> *data, const IECore::IntVectorData *indices, typename std::enable_if<!std::is_void<typename CortexTypeTraits<T>::USDType>::value>::type *enabler = nullptr ) const
```

Comment:
I take it this `enabler` wizardry is something to do with `IECore::Dispatch`

Reply:
Correct. `dispatch()` downcasts to the right data type and then calls the `VtValueFromExpandedData` functor. We use C++'s standard function overloading/matching rules to determine which function is called. The `enabler` shenanigans prevents this overload from being chosen for types which don't have an equivalent USD type.